app/src/main/java/com/example/paetz/yacguide/PartnersActivity.kt

```diff
@@ -92,44 +92,38 @@ class PartnersActivity : AppCompatActivity() {
     }
 
     private fun displayContent() {
-        title = "Kletterpartner"
+        title = getString(R.string.title_climbing_partner)
         val layout = findViewById<LinearLayout>(R.id.tableLayout)
         layout.removeAllViews()
         _checkboxMap.clear()
-        val partners = _db!!.partnerDao().all
-        val ascends = _db!!.ascendDao().all
+        val partners = _db.partnerDao().all
+        val ascends = _db.ascendDao().all
 
         // We need to sort the partners according to the number of ascends you have done with them
-        val ascendPartnerIds = ArrayList<Int>()
+        val ascendPartnerCount = SparseIntArray()
         for (ascend in ascends) {
-            ascendPartnerIds.addAll(ascend.partnerIds!!)
-        }
-        val sortedPartners = ArrayList<Partner>()
-        val partnerIdOccurences = ArrayList<Int>()
-        for (i in partners.indices) {
-            val occurenceCount = Collections.frequency(ascendPartnerIds, partners[i].id)
-            var index = i
-            for (j in 0 until i) {
-                if (occurenceCount > partnerIdOccurences[j]) {
-                    index = j
-                    break
+            ascend.partnerIds?.let {
+                for (id in it) {
+                    val prevValue = ascendPartnerCount.get(id, 0)
+                    ascendPartnerCount.put(id, prevValue + 1)
                 }
             }
-            sortedPartners.add(index, partners[i])
-            partnerIdOccurences.add(index, occurenceCount)
+        }
+
+        val sortedPartners = partners.sortedByDescending {
+            ascendPartnerCount.get(it.id, 0)
+        }.filter {
```

Comment:
Nice functional approach. I hope it works :D Have you tested it?

Reply:
Yes, tested it, works.